src/UrlRpcEncoder.ts

```diff
@@ -31,10 +31,21 @@ export class UrlRpcEncoder {
         if (!fragment.has('returnURL')) return null;
         const returnURL = fragment.get('returnURL')!;
         fragment.delete('returnURL');
-        const answerByPostMessage = returnURL === POSTMESSAGE_RETURN_URL
+
+        // guess the responseMethod in messages without one.
+        let responseMethod: ResponseMethod = ResponseMethod.GET;
+        if (fragment.has('responseMethod')) {
+            responseMethod = fragment.get('responseMethod')! as ResponseMethod;
+            fragment.delete('responseMethod');
+            if (!Object.values(ResponseMethod).includes(responseMethod)) {
+                throw new Error('Invalid ResponseMethod');
+            }
+        }
+
+        const answerByPostMessage = responseMethod === ResponseMethod.MESSAGE
                                     && (window.opener || window.parent);
```

Comment:
This check might lead to problems.

For example, when the response method is set indeed to `MESSAGE` in the request, but no opener or parent is availabe, then this check determines that we should NOT answer by postMessage. But the `responseMethod` is still assigned to the state as `MESSAGE`, so the State will still try to answer with postMessage when `reply()` is called.

Reply:
It is exactly the case we talked about earlier. Invoking the client with `ResponseMethod.POST_MESSAGE` will be rejected in case there is neither `window.opener` nor `window.parent`, as it is guaranteed to fail.